internal/venice-common/src/main/java/com/linkedin/venice/hadoop/schema/HDFSSchemaSource.java

```diff
@@ -121,6 +155,22 @@ void saveSchemaResponseToDisk(MultiSchemaResponse.Schema[] schemas, boolean isRm
     }
   }
 
+  void saveKeySchemaToDisk(SchemaResponse schema) throws IOException, IllegalStateException {
+    Preconditions.checkState(includeKeySchema, "Cannot be invoked with invalid key schema directory");
+    LOGGER.info("Caching key schema for store: {} in {}", storeName, keySchemaDir.getName());
+
+    Path schemaPath = new Path(keySchemaDir, String.valueOf(schema.getId()));
+    if (!fs.exists(schemaPath)) {
+      try (FSDataOutputStream outputStream = fs.create(schemaPath);
+          OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8)) {
+        outputStreamWriter.write(schema.getSchemaStr() + "\n");
+        outputStreamWriter.flush();
+      }
+    } else {
+      throw new IllegalStateException(String.format("The schema path %s already exists", schemaPath));
```

Comment:
In what scenario can this happen? Are we the ones responsible for cleaning up the schema paths after use?